src/openms/source/FORMAT/PercolatorInfile.cpp

```diff
@@ -76,6 +77,98 @@ namespace OpenMS
     return scan_identifier.removeWhitespaces();
   }
 
+  std::tuple<int,int> extractHighestChargeFromHeader_(const StringList& header)
+  {
+    int max_charge{INT_MIN}, min_charge{INT_MAX};
+    for (auto h : header)
+    {
+      if (h.hasPrefix("charge"))
+      {
+        h = h.substr(0, 6); // chop off "charge"
+        int z = h.toInt();
+        max_charge = z > max_charge ? z : max_charge;
+        min_charge = z < min_charge ? z : min_charge;
+      }
+    }
+    OPENMS_POSTCONDITION(max_charge != INT_MIN, "No charge column found!");
+    return make_tuple(min_charge, max_charge);
+  }
+
+  vector<PeptideIdentification> PercolatorInfile::load(const String& pin_file, bool higher_score_better, const String& score_name, String decoy_prefix)
+  {
+    CsvFile csv(pin_file, '\t');
+    StringList header;
+    csv.getRow(0, header);
+    unordered_map<String, size_t> to_idx; // map column name to column index
+    {
+      int idx{}; 
+      for (const auto& h : header) { to_idx[h] = idx++; }
+    }
+
+    auto [lowest_charge, highest_charge] = extractHighestChargeFromHeader_(header);
+    auto n_rows = csv.rowCount();
+    
+    vector<PeptideIdentification> pids(n_rows);
+    String spec_id;
+    size_t rank{1};
+    for (size_t i = 1; i != n_rows; ++i)
+    {
+      StringList row;
+      csv.getRow(i, row);
+      const String& sSpecId = row[to_idx.at("SpecId")];
+      if (sSpecId != spec_id)
+      {
+        pids.resize(pids.size() + 1);
+        pids.back().setHigherScoreBetter(higher_score_better);
+        pids.back().setScoreType(score_name);
+      }
+
+      int sScanNr = row[to_idx.at("ScanNr")].toInt();
+
+      String sPeptide = row[to_idx.at("Peptide")];
+      const double score = row[to_idx.at("score")].toDouble();
```

Comment:
TODO: there is no score column... we need to pass the column name for the main score